src/wallet/coincontrol.h

```diff
@@ -19,12 +21,12 @@ class CCoinControl
     bool fAllowOtherInputs;
     //! Includes watch only addresses which match the ISMINE_WATCH_SOLVABLE criteria
     bool fAllowWatchOnly;
-    //! Override estimated feerate
+    //! Override automatically calculated feerate, nFeeRate must be set if true
     bool fOverrideFeeRate;
-    //! Feerate to use if overrideFeeRate is true
-    CFeeRate nFeeRate;
-    //! Override the default confirmation target, 0 = use default
-    int nConfirmTarget;
+    //! Override the default payTxFee if set, 0 = use smart fee estimation
+    boost::optional<CFeeRate> m_feerate;
+    //! Override the default confirmation target
+    boost::optional<unsigned int> m_confirm_target;
```

Comment:
Hmm, maybe I shouldnt argue against boost::optional. I'm not a fan of optional types, but they do make sense here. Any further simplification would be nice, this stuff is gross (thanks for cleaning it up!).